application/backend/src/main/resources/apps/typerefinery/components/content/barchart/script.js

```diff
@@ -0,0 +1,152 @@
+function barChartComponent(component, id) {
+  Chart.defaults.global.defaultFontFamily = "Lato";
+  const dataSourceURL = component.getAttribute("data-source");
+  const defaultData = {
+    chartData: [75000, 75000, 75000, 15000, 14000, 12000, 11000, 11500, 11000],
+    labels: [
+      "Trazen:Win32/Qakbot",
+      "Ranson:Win32/Egre..",
+      "Quakbot",
+      "Android",
+      "Emotet",
+      "XAgentOSX",
+      "Pteranodon",
+      "PsExec",
+      "Minikatz",
+    ],
+    dataSetBorderColor: "#0099DE",
+    canvasBackgroundColor: "#001E3C",
+  };
+  const fetchData = async () => {
+    try {
+      const response = await fetch(dataSourceURL).then((res) => res.json());
+      !response ? drawBarChart(defaultData, id) : drawBarChart(response, id);
+    } catch (error) {
+      drawBarChart(defaultData, id);
+    }
+  };
+  fetchData();
+}
+
+function drawBarChart(barChartData, id) {
+  const {
+    labels,
+    chartData,
+    dataSetBorderColor = "#0099DE",
+    canvasBackgroundColor = "#001E3C",
+  } = barChartData;
+
+  const ctx = document.getElementById(`${id}-barChart`).getContext("2d");
+
+  // Canvas Background.
+  const plugin = {
+    id: "customCanvasBackgroundColor",
+    beforeDraw: (chart, args, options) => {
+      const { ctx } = chart;
+      ctx.save();
+      ctx.globalCompositeOperation = "destination-over";
+      ctx.fillStyle = "#001E3C" || "#99ffff";
+      ctx.fillRect(0, 0, chart.width, chart.height);
+      ctx.restore();
+    },
+  };
+
+  // Bar chart
+  new Chart(ctx, {
+    type: "horizontalBar",
+    data: {
+      labels: labels,
+      datasets: [
+        {
+          data: chartData,
+          backgroundColor: [
```

Comment:
Background colors can also come from the JSON file.

Reply:
done